scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+"""Rewrite a branch on top of a code style change commit so that all
+new commits on the branch look as if they were created after the
+code style change.
+"""
+
+# if workdir not clean exit
+#
+# git checkout oldbranch
+# git rebase codestyle_change_commit~1
+#
+# for oldbranch_commit in oldbranch
+#
+#     files = $(git diff --name-only oldbranch_commit oldbranch_commit~)
+#     git checkout oldbranch_commit -- $files
+#     code_style.fix($files)
+#     git add $files
+#     git commit --reuse-message=$(oldbranch_commit)
+
+import argparse
+import subprocess
+import sys
+from typing import List
+
+import code_style
+
+def working_directory_clean() -> bool:
+    """
+    Check if the working directory is clean (i.e. no changes to tracked
+    files staged or unstaged).
+    """
+    # Check for unstaged files
+    result = subprocess.run(["git", "diff", "--exit-code"], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        return False
+    # Check for staged files
+    result = subprocess.run(["git", "diff", "--cached", "--exit-code"], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        return False
+    return True
+
+def delete_branch(branch_name) -> bool:
+    """
+    Delete the given branch.
+    """
+    # Go to detached HEAD state in case we are deleting the checked-out branch
+    result = subprocess.run(["git", "checkout", "-d"], check=False)
+    if result.returncode != 0:
+        print("Error detaching HEAD.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "branch", "-D", branch_name], check=False)
+    if result.returncode != 0:
+        print("Error deleting branch '" + branch_name + "'.", file=sys.stderr)
+        return False
+    return True
+
+def rebase_on(branch: str, commit: str, new_branch: str) -> bool:
+    """
+    Make a new branch with the given name that consists of the given branch
+    rebased on the given commit.
+    """
+    # Create a new branch from the existing branch
+    result = subprocess.run(["git", "checkout", branch], check=False)
+    if result.returncode != 0:
+        print("Error checking out branch '"+branch+"'.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "checkout", "-b", new_branch], check=False)
+    if result.returncode != 0:
+        print("Error creating rebase destination branch '"+new_branch+"'.", \
+                file=sys.stderr)
+        return False
+
+    # Rebase the branch atop the given commit
+    result = subprocess.run(["git", "rebase", commit], check=False)
+    if result.returncode != 0:
+        print("Error rebasing branch '"+branch+"' on commit '"+commit+"'.", \
+                file=sys.stderr)
+        delete_branch(new_branch)
+        return False
+
+    return True
+
+def restyle_commit_onto_current(commit_hash: str) -> bool:
```

Comment:
Ultimately: this function returns true if everything went well and false otherwise. That's what exceptions are for!